### PR TITLE
Feat/comment section

### LINKED ISSUE #638: Comment a BP
Everyone can comment each BP, just like amazon commodity. So there are more information to support to more effective rating, then more EOS holders will join in rating.

### PR DESCRIPTION
**Pull request by hanakoori01**
### What does this PR do?

- Resolves #638 

### Steps to test

1. Run `make run`
2. Go to `localhost:3000`
3. Go to Block Producers > Block Producer Profile

### CODE REVIEW

File: webapp/src/routes/BlockProducers/BlockProducerRate.js [hunk 1]
```diff
@@ -217,6 +223,42 @@ const BlockProducerRate = () => {
     trustiness: parseFloat(formatNumber(rate?.trustiness || 0, 1))
   })
 
+  const sendComment = async () => {
+    try {
+      console.log(comment)
```
Comment (xavier506) [inline webapp/src/routes/BlockProducers/BlockProducerRate.js:228]: do we need this console.log?

Comment (xavier506) [inline webapp/src/routes/BlockProducers/BlockProducerRate.js:228]: do we need this console.log?

File: webapp/src/components/CommentCard/index.js [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+    if (loading || !data) {
+      return
+    }
+
+    const { comment } = data
+    const commentList = comment.map(comment => {
+      return { ...comment }
+    })
+    setComments([...commentList])
+    console.log(comments)
```
Comment (xavier506) [inline webapp/src/components/CommentCard/index.js:50]: do we need the console.log?

Comment (xavier506) [inline webapp/src/components/CommentCard/index.js:50]: do we need the console.log?

File: webapp/src/components/CommentCard/index.js [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+import FilterListIcon from '@mui/icons-material/FilterList'
+import Typography from '@mui/material/Typography'
+import { makeStyles } from '@mui/styles'
+import ThumbUpIcon from '@mui/icons-material/ThumbUp'
+import ThumbDownIcon from '@mui/icons-material/ThumbDown'
+import MenuItem from '@mui/material//MenuItem'
+import Menu from '@mui/material//Menu'
+import Moment from 'react-moment'
+import { useLazyQuery } from '@apollo/client'
+import { useSharedState } from '../../context/state.context'
```
Comment (tetogomez) [inline webapp/src/components/CommentCard/index.js:14]: move this lines to the second import group

Remember, the first import group is just for third party packages

File: webapp/src/components/CommentCard/index.js [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+  }
+  const [state] = useSharedState()
+  const [comments, setComments] = useState([])
+  const [getData, { loading, data }] = useLazyQuery(GET_COMMENTS, {
+    fetchPolicy: 'network-only'
+  })
+  const blockProducer = _get(producer, 'system.owner', null)
+
+  useEffect(() => {
+    if (loading || !data) {
```
Comment (tetogomez) [inline webapp/src/components/CommentCard/index.js:43]: use single line return

```
if (loading || !data) return
```

File: webapp/src/components/CommentCard/index.js [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+    setComments([...commentList])
+  }, [data, loading])
+
+  useEffect(() => {
+    const getDataAsync = async () => {
+      await getData({
+        variables: { bp: blockProducer }
+      })
+    }
+    if (!data) {
```
Comment (tetogomez) [inline webapp/src/components/CommentCard/index.js:60]: Add a blank space before `IF` declaration

File: webapp/src/components/CommentCard/index.js [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+    sendLike(transaction, like)
+  }
+
+  const handleClose = () => {
+    setAnchorEl(null)
+  }
+  const handleMenuItemClick = index => {
+    setSelectedIndex(index)
+    setAnchorEl(null)
+    if (index === 0) {
```
Comment (tetogomez) [inline webapp/src/components/CommentCard/index.js:75]: same here

File: webapp/src/components/CommentCard/index.js [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+      comments.sort(
+        (a, b) =>
+          new Date(...b.created_at.split('/').reverse()) -
+          new Date(...a.created_at.split('/').reverse())
+      )
+    }
+    if (index === 1) {
```
Comment (tetogomez) [inline webapp/src/components/CommentCard/index.js:82]: and here

File: webapp/src/components/CommentCard/index.js [hunk 8]
```diff
@@ -0,0 +1,240 @@
...
+      comments.sort((a, b) => (a.total_like < b.total_like ? 1 : -1))
+    }
+  }
+  useEffect(() => {
+    handleMenuItemClick(onloadstart, 0)
+  }, [])
+
+  const sendLike = async (id, like) => {
+    try {
+      if (!state.user?.accountName) {
```
Comment (tetogomez) [inline webapp/src/components/CommentCard/index.js:92]: single line return

```
if (!state.user?.accountName) return
```

File: webapp/src/components/CommentCard/index.js [hunk 9]
```diff
@@ -0,0 +1,240 @@
...
+        return
+      }
+      const transaction = {
```
Comment (tetogomez) [inline webapp/src/components/CommentCard/index.js:95]: add blank space after and before `const` declaration

File: webapp/src/components/CommentCard/index.js [hunk 10]
```diff
@@ -0,0 +1,240 @@
...
+                    mb={2}
+                    justifyContent='end'
+                  >
+                    <Moment fromNow>{comment.created_at}</Moment>
+                  </Typography>
+                </Box>
+                <Typography>{comment.content}</Typography>
+                <Box display='flex'>
+                  <Box justifyContent='space-between' className={classes.box}>
+                    {/* <Typography variant='body2'>
```
Comment (tetogomez) [inline webapp/src/components/CommentCard/index.js:186]: we need this commented. code ?

File: webapp/src/components/CommentCard/index.js [hunk 11]
```diff
@@ -0,0 +1,240 @@
...
+                        Trustiness: {comment.trustiness}
+                      </Typography> */}
+                  </Box>
+                  <Box justifyContent='end' width='20%' className={classes.box}>
+                    <Typography
+                      variant='body3'
+                      textTransform='uppercase'
+                      mr={1}
+                    >
+                      Is this helpful?
```
Comment (tetogomez) [inline webapp/src/components/CommentCard/index.js:208]: move this text into i18n json files

File: webapp/src/components/MenuDropDown/index.js [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+  function handleListKeyDown(event) {
+    if (event.key === 'Tab') {
+      event.preventDefault()
+      setOpen(false)
+    } else if (event.key === 'Escape') {
+      setOpen(false)
+    }
+  }
+
+  // return focus to the button when we transitioned from !open -> open
```
Comment (tetogomez) [inline webapp/src/components/MenuDropDown/index.js:36]: delete this comment

File: webapp/src/routes/BlockProducers/BlockProducerRate.js [hunk 13]
```diff
@@ -217,6 +222,40 @@ const BlockProducerRate = () => {
...
   })
 
+  const sendComment = async () => {
+    try {
+      if (!state.user?.accountName) {
+        setShowMessage(true)
+
+        return
+      }
+      const transaction = {
```
Comment (tetogomez) [inline webapp/src/routes/BlockProducers/BlockProducerRate.js:232]: Blank space after and before `const` declaration

File: webapp/src/routes/BlockProducers/BlockProducerRate.js [hunk 14]
```diff
@@ -595,18 +596,93 @@ const BlockProducerRate = () => {
...
+                <Typography className={classes.commentTitle} variant='h6'>
+                  Add Comments
+                </Typography>
+                <TextField
+                  className={classes.textField}
+                  id='outlined-multiline-static'
+                  multiline
+                  rows={4}
+                  placeholder='Comment on your rating reasoning.
+          Please be respectful.'
```
Comment (tetogomez) [inline webapp/src/routes/BlockProducers/BlockProducerRate.js:615]: check the indentation for this text

**Review by tetogomez — COMMENTED** [review]
Good job @hanakoori01 

Please check the comments

**Pull request merged**